### DISCUSSION TITLE
TS errors when referencing tokens in template literals?

### DISCUSSION THREAD

**Discussion by doksara** [chakra-ui/panda#2733] [General]
I recently set up my NextJS with PandaCSS. Since I come from `styled-components`, I got used to using template literals syntax to define my styles. In the [official docs](https://panda-css.com/docs/concepts/template-literals#using-tokens), it says that I can just reference my token like this:

```
import { css } from '../styled-system/css'
 
const className = css`
  font-size: {fontSizes.md};
  font-weight: token(fontWeights.bold, 700);
`
```

and it works just fine. But the problem is, I am getting TS errors everywhere:
<img width="454" alt="Screenshot 2024-07-16 at 16 21 23" src="https://github.com/user-attachments/assets/9764ea97-558e-4ee2-ace4-d04b08c6f202">

How do I fix this? I can't find anything related to TS setup in the official documentation.

**Comment by segunadebayo** [comment]
This isn't a TS Error, per se. It's an error from the ts-styled-plugin. I would recommend the following:
- Use the `{colors.primaryStone}` syntax
- Consider switching to `vscode-styled-components` plugin instead

**Reply by doksara** [reply]
I am still getting an error from the plugin:
<img width="432" alt="Screenshot 2024-07-17 at 13 24 44" src="https://github.com/user-attachments/assets/527c90e0-b9f8-40a7-8734-795d438f877e">

Also, I'm already using the `vscode-styled-components` plugin.

**Reply by segunadebayo** [reply]
Panda's token reference syntax is incompatible with styled-components, so I recommend not using that ts-styled plugin or creating an ignore rule.
